tests/test_arrow_dataset.py

```diff
@@ -1850,6 +1851,83 @@ def test_with_transform(self, in_memory):
                     self.assertNotEqual(dset.format, dset2.format)
                     self.assertNotEqual(dset._fingerprint, dset2._fingerprint)
 
+    def test_task_text_classification(self, in_memory):
+        labels = sorted(["pos", "neg"])
+        features_before_cast = Features(
+            {
+                "input_text": Value("string"),
+                "input_labels": Value("int32"),
+            }
+        )
+        features_after_cast = Features(
+            {
+                "text": Value("string"),
+                "labels": ClassLabel(names=labels),
+            }
+        )
+        info = DatasetInfo(
+            features=features_before_cast,
+            task_templates=TextClassification(text_column="input_text", label_column="input_labels", labels=labels),
+        )
+        data = {"input_text": ["i love transformers!"], "input_labels": [1]}
+        with tempfile.TemporaryDirectory() as tmp_dir, Dataset.from_dict(data, info=info) as dset:
+            dset = self._to(in_memory, tmp_dir, dset)
+            self.assertSetEqual(set(["input_text", "input_labels"]), set(dset.column_names))
+            self.assertDictEqual(features_before_cast, dset.features)
+            dset = dset.prepare_for_task(task="text-classification")
+            self.assertSetEqual(set(["labels", "text"]), set(dset.column_names))
+            self.assertDictEqual(features_after_cast, dset.features)
+
+    def test_task_question_answering(self, in_memory):
+        features_before_cast = Features(
+            {
+                "input_context": Value("string"),
+                "input_question": Value("string"),
+                "input_answers": Sequence(
+                    {
+                        "text": Value("string"),
+                        "answer_start": Value("int32"),
+                    }
+                ),
+            }
+        )
+        features_after_cast = Features(
+            {
+                "context": Value("string"),
+                "question": Value("string"),
+                "answers": Sequence(
+                    {
+                        "text": Value("string"),
+                        "answer_start": Value("int32"),
+                    }
+                ),
+            }
+        )
+        info = DatasetInfo(
+            features=features_before_cast,
+            task_templates=QuestionAnswering(
+                context_column="input_context", question_column="input_question", answers_column="input_answers"
+            ),
+        )
+        data = {
+            "input_context": ["huggingface is going to the moon!"],
+            "input_question": ["where is huggingface going?"],
+            "input_answers": [{"text": ["to the moon!"], "answer_start": [2]}],
+        }
+        with tempfile.TemporaryDirectory() as tmp_dir, Dataset.from_dict(data, info=info) as dset:
```

Comment:
for some reason, using a context manager for `dset` produces an error on the windows tests:

```
tests\test_arrow_dataset.py:1879: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\tools\miniconda3\lib\tempfile.py:809: in __exit__
    self.cleanup()
C:\tools\miniconda3\lib\tempfile.py:813: in cleanup
    _shutil.rmtree(self.name)
C:\tools\miniconda3\lib\shutil.py:500: in rmtree
    return _rmtree_unsafe(path, onerror)
C:\tools\miniconda3\lib\shutil.py:395: in _rmtree_unsafe
    onerror(os.unlink, fullname, sys.exc_info())
```

Reply:
update: i've now added context management for all `dset` objects in the tests. hopefully the tests will now pass 🤞